p2p/host.go

```diff
@@ -413,6 +422,9 @@ func New(
 	if err != nil {
 		return nil, fmt.Errorf("failed to initialize libp2p host: %w", err)
 	}
+	if identifyConn == nil {
+		panic("BUG: identify service not set")
+	}
```

Comment:
Why panic instead of error?

Reply:
In case if `identifyConn` is not set there's definitely a bug, e.g. in `libp2p` or `fx` library.
Panicking in case of an internal error (bug) makes sense IMO

Reply:
Removed this panic, `p2p.New()` now returns an error if `IDService` cannot be discovered